src/driveformatthread.cpp

```diff
@@ -135,50 +135,26 @@ void DriveFormatThread::run()
 
 #elif defined(Q_OS_LINUX)
 
-    if (::access(_device.constData(), W_OK) != 0)
-    {
-        /* Not running as root, try to outsource formatting to udisks2 */
-        if(_label.isEmpty())
-        {
-#ifndef QT_NO_DBUS
-            UDisks2Api udisks2;
-            if (udisks2.formatDrive(_device))
-            {
-                emit success();
-            }
-            else
-            {
-#endif
-                emit error(tr("Error formatting (through udisks2)"));
-#ifndef QT_NO_DBUS
-            }
-#endif
-        }
-        else
-        {
-            emit error(tr("Elevated privileges needed to properly format drive."));
-        }
-        return;
-    }
-
-
     QProcess proc;
     QByteArray partitionTable;
     QStringList args;
+    QStringList args2;
     QByteArray fatpartition = _device;
     partitionTable = "8192,,0E\n"
             "0,0\n"
             "0,0\n"
             "0,0\n";
-    args << "-uS" << _device;
+    args << "sfdisk" << "-uS" << "--no-reread" << _device;
     if (isdigit(fatpartition.at(fatpartition.length()-1)))
         fatpartition += "p1";
     else
         fatpartition += "1";
 
     unmount_disk(_device);
+    
     proc.setProcessChannelMode(proc.MergedChannels);
-    proc.start("sfdisk", args);
+  
+    proc.start("pkexec", args);

```

Comment:
Swapped to a single exec of pkexec instead of sfdisk? Any downsides to this approach?

Reply:
pkexec was an attempt to give sfdisk elevated permissions - it's been reverted now.